### DISCUSSION TITLE
What is "These operation-level options override the global server options." in Swagger.

### DISCUSSION THREAD

**Discussion by gbaian10** [fastapi/fastapi#11943] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.post("/hello/")
def create_item() -> str:
    return "Hello, FastAPI!"
```

### Description

When I am in fastapi >= 0.111.1 version, the same code will generate an extra item in swagger.
I am not sure what its role is, but this thing will not exist in version 0.111.0.

![image](https://github.com/user-attachments/assets/29729f17-ac33-427f-9a8e-9f189ea6df98)

![image](https://github.com/user-attachments/assets/b6b70f7e-32a9-441b-a7fa-ff2e9afa09a3)

### Operating System

Linux, Windows

### FastAPI Version

0.111.1 or 0.112.0

### Pydantic Version

2.8.2

### Python Version

3.12.4

**Comment by n8sty** [comment, marked as answer]
This change https://github.com/fastapi/fastapi/pull/11459/files was introduced in 0.111.1 (https://github.com/fastapi/fastapi/releases/tag/0.111.1) which bumped the version of Swagger to whatever the latest 5.x release is (it's definitely beyond 5.9.x). That is the why you are seeing differences in the auto-generated docs.

The what does this bit of documentation mean can be answered by saying it's a bug in OAS 3.1. I believe this is the relevant issue on the Swagger UI project https://github.com/swagger-api/swagger-ui/issues/9947.

**Reply by gbaian10** [reply]
Thank you for your reply, although I still don't know how to deal with it, or what other effects it will have, but at least I know the source of the problem.

Thanks again for your reply.

**Comment by xzfxzrs** [comment]
today i met the same problem.

then that is my solution :

first, close default swagger by
```
app = FastAPI(
    docs_url=None,  # close default swagger-ui
    ...
)
```

then add a custom doc endpoint

```
@app.get("/docs", include_in_schema=False)
async def custom_swagger_ui_html():
    """custom swagger ui"""
    if not app.openapi_url:
        raise HTTPException(status_code=404)

    html_response = get_swagger_ui_html(
        openapi_url=app.openapi_url,
        title=app.title + " - Swagger UI",
        oauth2_redirect_url=app.swagger_ui_oauth2_redirect_url,
        swagger_ui_parameters=app.swagger_ui_parameters,
    )
    # inject on complete method, in the function add custom css in head
    new_body = html_response.body.decode().replace(
        r"const ui = SwaggerUIBundle({",
        r"""
        const ui = SwaggerUIBundle({
            "onComplete": () => {
                // create style element
                var style = document.createElement('style');

                // add style for .operation-servers
                style.innerHTML = `
                .operation-servers {
                    display: none
                }
                `;

                // add style element into head 
                document.head.appendChild(style);
            },
        """,
    )
    return HTMLResponse(content=new_body)
```

hope this is helpful for you
